tsl/src/compression/compression.c

```diff
@@ -1180,15 +1091,19 @@ row_compressor_update_group(RowCompressor *row_compressor, TupleTableSlot *row)
 	 * many segmentby columns.
 	 *
 	 */
-	row_compressor->sequence_num =
-		get_sequence_number_for_current_group(row_compressor->compressed_table,
-											  row_compressor->index_oid,
-											  row_compressor->uncompressed_col_to_compressed_col,
-											  row_compressor->per_column,
-											  row_compressor->n_input_columns,
-											  AttrOffsetGetAttrNumber(
+	if (!row_compressor->recompressing)
+		row_compressor->sequence_num =
+			get_sequence_number_for_current_group(row_compressor->compressed_table,
+												  row_compressor->index_oid,
 												  row_compressor
-													  ->sequence_num_metadata_column_offset));
+													  ->uncompressed_col_to_compressed_col,
+												  row_compressor->per_column,
+												  row_compressor->n_input_columns,
+												  AttrOffsetGetAttrNumber(
+													  row_compressor
+														  ->sequence_num_metadata_column_offset));
+	else
+		row_compressor->sequence_num = SEQUENCE_NUM_GAP;
```

Comment:
do we have a test that shows the sequence number is increasing in recompressed segments? dont we need to increment this?